lib/std/fmt.zig

```diff
@@ -1503,8 +1503,16 @@ fn formatDuration(data: FormatDurationData, comptime fmt: []const u8, options: s
 
     // worst case: "-XXXyXXwXXdXXhXXmXX.XXXs".len = 24
     var buf: [24]u8 = undefined;
-    var fbs = std.io.fixedBufferStream(&buf);
+
+    return tryFormatDuration(data, &buf, options, writer) catch |err| switch (err) {
```

Comment:
This makes the test below fail because you catch the error.NoSpaceLeft from the external writer, too
```zig
 var buf: [1]u8 = undefined; 
 std.testing.expectError(error.NoSpaceLeft, std.fmt.bufPrint(&buf, "{}", .{std.fmt.fmtDuration(10000)}));
```

Reply:
Good call. I'll have a go at fixing it :-)